File: server-ruby-sinatra/api/offer_status_api.rb
require 'json'


MyApp.add_route('POST', '/api/{version}/offer/status/create', {
  "resourcePath" => "/OfferStatus",
  "summary" => "Create Offer Status",
  "nickname" => "create_offer_transaction_status",
  "responseClass" => "OfferTransactionStatusResponse",
  "endpoint" => "/api/{version}/offer/status/create",
  "notes" => "Create an offer status record",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "name",
      "description" => "The name of the status",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "The description of the status",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "code",
      "description" => "The status code, must be unique ",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "role",
      "description" => "The actor that uses the status",
      "dataType" => "String",
      "allowableValues" => "",
      "defaultValue" => "'ANY'",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "The active status",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "true",
      "paramType" => "query",
    },
    {
      "name" => "application_ids",
      "description" => "The applications to associate the status with, if null then for all.",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/offer/status/delete', {
  "resourcePath" => "/OfferStatus",
  "summary" => "Delete Offer Status",
  "nickname" => "delete_offer_transaction_status",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/api/{version}/offer/status/delete",
  "notes" => "Mark an offer status record as deleted",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "status_id",
      "description" => "The id of the record to delete",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/offer/status/get', {
  "resourcePath" => "/OfferStatus",
  "summary" => "Get Offer Status",
  "nickname" => "get_offer_transaction_status",
  "responseClass" => "OfferTransactionStatusResponse",
  "endpoint" => "/api/{version}/offer/status/get",
  "notes" => "Get an offer status record",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "status_id",
      "description" => "The id of the record to get ",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/offer/status/search', {
  "resourcePath" => "/OfferStatus",
  "summary" => "Search Offer Status",
  "nickname" => "search_offer_transaction_statuses",
  "responseClass" => "Array<OfferTransactionStatusResponse>",
  "endpoint" => "/api/{version}/offer/status/search",
  "notes" => "Search for the available offer statuses",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "Search the name and description fields",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "role",
      "description" => "The actors role, possible values are: CUSTOMER, RETAILER, or ANY",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "The application making the request",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "The field to sort on, possible values are: ID, CREATED, UPDATED, DELETED, SEARCH_TAGS, ACTIVE, NAME, DESCRIPTION, CODE, ROLE",
      "dataType" => "String",
      "allowableValues" => "[ID, CREATED, UPDATED, DELETED, SEARCH_TAGS, ACTIVE, NAME, DESCRIPTION, CODE, ROLE]",
      "defaultValue" => "'CODE'",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "Determines whether the results are in descending order",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "true",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The start index for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "0",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The limit for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "20",
      "paramType" => "query",
    },
    {
      "name" => "include_inactive",
      "description" => "If true include inactive items",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "false",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/offer/status/update', {
  "resourcePath" => "/OfferStatus",
  "summary" => "Update Offer Status",
  "nickname" => "update_offer_transaction_status",
  "responseClass" => "OfferTransactionStatusResponse",
  "endpoint" => "/api/{version}/offer/status/update",
  "notes" => "Update an offer status record",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "Used to update the user&#39;s current location",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "status_id",
      "description" => "The id of the record to update ",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "name",
      "description" => "The name of th status",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "The description of the status",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "code",
      "description" => "The status code, must be unique ",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "role",
      "description" => "The actor that uses the status",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "The active status",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "application_ids",
      "description" => "The applications to associate the status with, if null then for all.",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

File: server-scala-play-server/app/api/OfferStatusApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.OfferTransactionStatusResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait OfferStatusApi {
  /**
    * Create Offer Status
    * Create an offer status record
    * @param name The name of the status
    * @param code The status code, must be unique 
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param latitude Used to update the user&#39;s current location
    * @param longitude Used to update the user&#39;s current location
    * @param description The description of the status
    * @param role The actor that uses the status
    * @param active The active status
    * @param applicationIds The applications to associate the status with, if null then for all.
    */
  def createOfferTransactionStatus(version: BigDecimal, name: String, code: Int, deviceId: Option[String], accountId: Option[Long], latitude: Option[Double], longitude: Option[Double], description: Option[String], role: Option[String], active: Option[Boolean], applicationIds: Option[String]): OfferTransactionStatusResponse

  /**
    * Delete Offer Status
    * Mark an offer status record as deleted
    * @param statusId The id of the record to delete
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param latitude Used to update the user&#39;s current location
    * @param longitude Used to update the user&#39;s current location
    */
  def deleteOfferTransactionStatus(version: BigDecimal, statusId: Long, deviceId: Option[String], accountId: Option[Long], latitude: Option[Double], longitude: Option[Double]): SirqulResponse

  /**
    * Get Offer Status
    * Get an offer status record
    * @param statusId The id of the record to get 
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param latitude Used to update the user&#39;s current location
    * @param longitude Used to update the user&#39;s current location
    */
  def getOfferTransactionStatus(version: BigDecimal, statusId: Long, deviceId: Option[String], accountId: Option[Long], latitude: Option[Double], longitude: Option[Double]): OfferTransactionStatusResponse

  /**
    * Search Offer Status
    * Search for the available offer statuses
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param latitude Used to update the user&#39;s current location
    * @param longitude Used to update the user&#39;s current location
    * @param keyword Search the name and description fields
    * @param role The actors role, possible values are: CUSTOMER, RETAILER, or ANY
    * @param appKey The application making the request
    * @param sortField The field to sort on, possible values are: ID, CREATED, UPDATED, DELETED, SEARCH_TAGS, ACTIVE, NAME, DESCRIPTION, CODE, ROLE
    * @param descending Determines whether the results are in descending order
    * @param start The start index for pagination
    * @param limit The limit for pagination
    * @param includeInactive If true include inactive items
    */
  def searchOfferTransactionStatuses(version: BigDecimal, deviceId: Option[String], accountId: Option[Long], latitude: Option[Double], longitude: Option[Double], keyword: Option[String], role: Option[String], appKey: Option[String], sortField: Option[String], descending: Option[Boolean], start: Option[Int], limit: Option[Int], includeInactive: Option[Boolean]): List[OfferTransactionStatusResponse]

  /**
    * Update Offer Status
    * Update an offer status record
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param latitude Used to update the user&#39;s current location
    * @param longitude Used to update the user&#39;s current location
    * @param statusId The id of the record to update 
    * @param name The name of th status
    * @param description The description of the status
    * @param code The status code, must be unique 
    * @param role The actor that uses the status
    * @param active The active status
    * @param applicationIds The applications to associate the status with, if null then for all.
    */
  def updateOfferTransactionStatus(version: BigDecimal, deviceId: Option[String], accountId: Option[Long], latitude: Option[Double], longitude: Option[Double], statusId: Option[Long], name: Option[String], description: Option[String], code: Option[Int], role: Option[String], active: Option[Boolean], applicationIds: Option[String]): OfferTransactionStatusResponse
}
